07-React/react-boilerplate-master/components/button.js

```diff
@@ -0,0 +1,26 @@
+var React = require ('react');
+
+var Button = React.createClass({
+
+    propTypes: {
+        onClick: React.PropTypes.func.isRequired
+    },
+
+    render: function () {
+        return (
+            <button {...this.getProps()}>{this.props.children}</button>
+        );
+    },
+
+    getProps: function () {
+        return {
+            onClick: this.handleOnClick
+        }
+    },
+
+    handleOnClick: function (event) {
+        this.props.onClick(event);
```

Comment:
if you remove the required from the propTypes this should be validated before using it